### PR TITLE
protoc-gen-go-grpc: add requirement of embedding UnimplementedServer in services

### PR DESCRIPTION
**Pull request by dfawley**

### CODE REVIEW

**Comment by dfawley** [comment]
Good idea.  I went with `mustEmbedUnimplementedFooServer`.  PTAL.

File: balancer/grpclb/grpc_lb_v1/load_balancer_grpc.pb.go [hunk 1]
```diff
@@ -61,18 +61,22 @@ func (x *loadBalancerBalanceLoadClient) Recv() (*LoadBalanceResponse, error) {
 }
 
 // LoadBalancerServer is the server API for LoadBalancer service.
+// All implementations must embed UnimplementedLoadBalancerServer
+// for forward compatibility
 type LoadBalancerServer interface {
 	// Bidirectional rpc to get a list of servers.
 	BalanceLoad(LoadBalancer_BalanceLoadServer) error
+	mustEmbedUnimplementedLoadBalancerServer()
```
Comment (menghanl) [inline balancer/grpclb/grpc_lb_v1/load_balancer_grpc.pb.go:69]: Oh, we said we want to revert the changes to the generated code, because they are API changes.

File: cmd/protoc-gen-go-grpc/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# protoc-gen-go-grpc
+
+This tool generates Go language bindings of `service`s in protobuf definition
+files for gRPC.  For usage information, please see our [quick start
+guide](https://grpc.io/docs/languages/go/quickstart/).
```
Comment (menghanl) [inline cmd/protoc-gen-go-grpc/README.md:5]: Quick start doesn't have examples for using this plugin (yet?)

Even if it does, it would be helpful to have a copy-paste ready command here?
Or point to https://github.com/grpc/grpc-go/blob/master/regenerate.sh#L60

File: cmd/protoc-gen-go-grpc/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+By default, to register services using the methods generated by this tool, the
```
[the remark this answers is not in the document]
Comment (dfawley) [reply cmd/protoc-gen-go-grpc/README.md:9]: Done.

File: cmd/protoc-gen-go-grpc/README.md (hunk 2 above)
Comment (dfawley) [reply cmd/protoc-gen-go-grpc/README.md:5]: Documentation is very much a work in progress.  Let's make incremental progress on it until we eventually release the tool at 1.0.
